src/keylessBackup/saga.ts

```diff
@@ -110,8 +118,48 @@ function* handleKeylessBackupRestore({
     encryptionAddress,
   })
 
-  // TODO(ACT-778): use mnemonic
-  yield* call(decryptPassphrase, torusKeyshareBuffer, valoraKeyshareBuffer, encryptedMnemonic)
+  const decryptedMnemonic = yield* call(
+    decryptPassphrase,
+    torusKeyshareBuffer,
+    valoraKeyshareBuffer,
+    encryptedMnemonic
+  )
+
+  const { privateKey } = yield* call(generateKeysFromMnemonic, decryptedMnemonic)
+  if (!privateKey) {
+    throw new Error('Failed to convert mnemonic to hex')
+  }
+  const backupAccount = privateKeyToAddress(privateKey)
+  if (!(yield* call(walletHasBalance, backupAccount))) {
+    // show zero balance modal
+    yield* put(keylessBackupShowZeroBalance())
+
+    // wait for user to click continue or bail from keyless backup
+    const [_continueAction, bailAction] = yield* race([
+      take(keylessBackupAcceptZeroBalance.type),
+      take(keylessBackupBail.type),
+    ])
+    if (bailAction) {
+      // TODO(ACT-874): Navigate to ImportSelect screen
+      return
+    }
+  }
+
+  const account: string | null = yield* call(
+    assignAccountFromPrivateKey,
+    privateKey,
+    decryptedMnemonic
+  )
+  if (!account) {
+    throw new Error('Failed to assign account from private key')
+  }
+  // Set key in phone's secure store
+  yield* call(storeMnemonic, decryptedMnemonic, account)
+
+  yield* put(refreshAllBalances())
```

Comment:
is this required? seems like initializeAccountSaga also puts it

Reply:
I don't think so. Its in the normal import flow so I kept it, but I think you're right its not necessary